coalib/testing/BaseTestHelper.py

```diff
@@ -0,0 +1,44 @@
+from coala_utils.Comparable import Comparable
+
+
+class BaseTestHelper:
+    """
+    This is a base class for all Bears' tests of coala's testing API.
+    """
+
+    def assert_comparable_objects(self,
+                                  observed_result,
+                                  expected_result):
+        """
+        Compares individual fields of the result objects and yields messages
+        in case of an attribute mismatch.
```

Comment:
'attribute mismatch' is unclear.

Is it attribute *type* mismatch, or attribute *value* mismatch?

When I read `assert_comparable_objects`, I assume it means it checks that the *types* are comparable, but does not check that they are the same.

If it is about the values, then the name should include they key word `equals`, like [`assertCountEqual`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertCountEqual)

However, by reading the code I can see that `comparable` here refers to `coala_utils.Comparable`.  That needs to be described in the docstring.

fwiw, most of this logic should be put into `Comparable.__eq__`, and similar methods.

Then the code here would simple use `!=` , and raise asserts accordingly.